delta-api/src/main/java/io/cdap/delta/api/DDLEvent.java

```diff
@@ -69,6 +71,22 @@ public boolean equals(Object o) {
       Objects.equals(schema, ddlEvent.schema) &&
       Objects.equals(primaryKey, ddlEvent.primaryKey);
   }
+  @Override
+  public String toString() {
+    String operationString = "operation=" + operation;
+    String isSnapshotString = "isSnapshot=" + this.isSnapshot();
+    String changeTypeString = "changeType=" + this.getChangeType();
+    String primaryKeyString = primaryKey == null || primaryKey.isEmpty() ?
+                              null : "primaryKey=" + "[" + String.join(",", primaryKey) + "]";
+    String offsetString = this.getOffset().get() != null ? "offset=" + this.getOffset().get() : null;
+    String sourceTimeString  = this.getSourceTimestampMillis() != null ? "sourceTimestampMillis=" +
+                              this.getSourceTimestampMillis() : null;
+    String ddlEvent = Stream.of(changeTypeString, operationString, primaryKeyString, offsetString,
+                                    isSnapshotString, sourceTimeString)
+                            .filter(s -> s != null && !s.isEmpty())
+                            .collect(Collectors.joining(", "));
+    return "{" + ddlEvent + "}";
```

Comment:
1. We can use StringJoiner instead which supports delimeter, oprefix and suffix
2. We should implement toString in Offset class instead of fetching the map here
3. We can directly rely on List's toString for primary key

```suggestion
    StringJoiner stringJoiner = new StringJoiner(",", "{", "}");
    stringJoiner.add("operation=" + operation);
    stringJoiner.add("isSnapshot=" + this.isSnapshot());
    stringJoiner.add("changeType=" + this.getChangeType());
    if (primaryKey != null && !primaryKey.isEmpty()) {
      stringJoiner.add("primaryKey=" + primaryKey);
    }
    if (this.getOffset() != null) {
      stringJoiner.add("offset=" + this.getOffset());
    }
    if (this.getSourceTimestampMillis() != null) {
      stringJoiner.add("sourceTimestampMillis=" + this.getSourceTimestampMillis());
    }
    return stringJoiner.toString();
```